### PR TITLE
`v11.x.x` Radio updates

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/22

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by gwyneplaine** [comment]
Also changed the --pc-choice-size token to accomodate the new radio height (16px rather than 18px) 
before: 
Screenshot 2023-05-24 at 9 35 46 pm

after:
Screenshot 2023-05-24 at 9 32 24 pm

File: polaris-react/src/components/Choice/Choice.scss [hunk 1]
```diff
@@ -4,6 +4,7 @@
   display: flex;
   justify-content: flex-start;
   padding: var(--p-space-1) 0;
+  align-items: center;
```
Comment (jesstelford) [inline polaris-react/src/components/Choice/Choice.scss]: Does this need to be wrapped in the summer editions flag?

```suggestion
  #{$se23} & {
    align-items: center;
  }
```

File: polaris-react/src/components/Choice/Choice.scss [hunk 2]
```diff
@@ -74,6 +83,10 @@
 .Descriptions {
   // stylelint-disable-next-line -- Polaris component custom properties
   --pc-choice-size: 20px;
+  #{$se23} & {
+    // stylelint-disable-next-line -- Polaris component custom properties
```
Comment (jesstelford) [inline polaris-react/src/components/Choice/Choice.scss]: Is `PSE23` a standard / common identifier we're using for finding these lint rule disablings? If so, this one is missing:

```suggestion
    // stylelint-disable-next-line -- PSE23 Polaris component custom properties
```

File: polaris-react/src/components/RadioButton/RadioButton.scss [hunk 3]
```diff
@@ -20,30 +24,82 @@
...
+    --p-color-border-input: rgba(138, 138, 138, 1);
+    // stylelint-disable-next-line -- PSE23 temporary property overrides
+    --p-color-bg: #fff;
+    // stylelint-disable-next-line -- PSE23 temporary property overrides
+    --p-color-bg-interactive: #fff;
+    // stylelint-disable-next-line -- PSE23 temporary property overrides
+    --p-border-width-1: 0.66px;
+    border: var(--p-border-width-1) solid var(--p-color-border-input);
+  }
+  // stylelint-disable-next-line -- PSE23 temporary property overrides
```
Comment (jesstelford) [inline polaris-react/src/components/RadioButton/RadioButton.scss]: Looks like this one might have gotten incorrectly reworded?

```suggestion
  // stylelint-disable-next-line -- Polaris component custom properties
```

File: polaris-react/src/components/Choice/Choice.scss (hunk 1 above)
Comment (gwyneplaine) [reply polaris-react/src/components/Choice/Choice.scss]: This one was a good samaritan fix and can be applied to both se23 and the old code, but yeah good call to throw this behind the flag 👍

File: polaris-react/src/components/Choice/Choice.scss [hunk 4]
```diff
@@ -41,13 +45,29 @@
...
+    color: var(--p-color-text-disabled);
+  }
+}
+
 .Control {
   // stylelint-disable-next-line -- Polaris component custom properties
   --pc-choice-size: 20px;
+  #{$se23} & {
+    // stylelint-disable-next-line -- PSE23 temporary property overrides
+    --pc-choice-size: 18px;
```
Comment (gwyneplaine) [inline polaris-react/src/components/Choice/Choice.scss:57]: This results in a choice size of 16px (as per the designs) with a 1px offset

File: polaris-react/src/components/Choice/Choice.scss [hunk 5]
```diff
@@ -41,13 +45,29 @@
...
+  }
 
   @media #{$p-breakpoints-md-down} {
     // stylelint-disable-next-line -- Polaris component custom properties
     --pc-choice-size: 22px;
+    #{$se23} {
+      // stylelint-disable-next-line -- PSE23 Polaris component custom properties
+      --pc-choice-size: 20px;
```
Comment (gwyneplaine) [inline polaris-react/src/components/Choice/Choice.scss:65]: As discussed for mdDown we want the element to be slightly larger (18px)

**Review by jesstelford — APPROVED** [review]

**Comment by bernardojoaogarcia** [comment]
Looks great! 🚀 We also made the helper text `disabled` for the radio buttons' disabled state. Was that included in this update?

**Review by bernardojoaogarcia — APPROVED** [review]

**Pull request merged**